### PR TITLE
added BSC chain id to encodeSigHeader

### PR DESCRIPTION
**Pull request by Ezkerrox**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added entries in `PENDING.md` with issue # 
- [ ] rereviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

Explained at #306

### CODE REVIEW

File: bsc/header.go [hunk 1]
```diff
@@ -187,13 +187,14 @@ func Keccak256(data ...[]byte) []byte {
 // SealHash returns the hash of a block prior to it being sealed.
 func SealHash(header *Header) (hash Hash) {
 	hasher := sha3.NewLegacyKeccak256()
-	encodeSigHeader(hasher, header)
+	encodeSigHeader(hasher, header, big.NewInt(56))
```
Comment (forcodedancing) [inline bsc/header.go:190]: Need to a way to distinguish testnet and mainnet, i think.

**Comment by Ezkerrox** [comment]
Set chain ID as ScKeeper.BscSideChainId(ctx), is this correct? Also updated header_test.go

**Comment by Ezkerrox** [comment]
> 

BscSideChainId is just a string containing bsc, already updated. Is there a need to add testnet also? tests already updated and working

**Comment by forcodedancing** [comment]
> > 
> 
> BscSideChainId is just a string containing bsc, already updated. Is there a need to add testnet also? tests already updated and working @forcodedancing
Thanks a lot @Ezkerrox for the contribution, we will look into it in details and conduct more tests accordingly.

**Comment by Ezkerrox** [comment]
> > > 
> > 
> > 
> > BscSideChainId is just a string containing bsc, already updated. Is there a need to add testnet also? tests already updated and working @forcodedancing
> > Thanks a lot @Ezkerrox for the contribution, we will look into it in details and conduct more tests accordingly.

Thanks for your reply:) I am glad to help @forcodedancing. This is king of a big issue, If I am not mistaken, right now slashing for double signing on BSC its not working. This mechanism its really important to keep BSCs safety, so I guess it should be solved asap. Thank you for taking care :)

**Pull request closed without merging**